api/class2.js

```diff
@@ -1,79 +1,202 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
 
-const filename = 'database/database.sqlite';
-const sqlite3 = require('sqlite3').verbose();
+const filename = "database/database.sqlite";
+const sqlite3 = require("sqlite3").verbose();
 let db = new sqlite3.Database(filename);
 // db.run("PRAGMA foreign_keys = ON");
-
-router.get('/customers', function(req, res) {
-  var sql = 'select * from customers';
-
-  db.all(sql, [], (err, rows ) => {
+router.get("/customers", function(req, res) {
+  var sql = "select * from customers";
+  db.all(sql, [], (err, rows) => {
     res.status(200).json({
       customers: rows
     });
   });
 });
 
-
-router.get('/customers/:id', function(req, res) {
+router.get("/customers/:id", function(req, res) {
   // TODO: add code here
+  var sql = `select * from customers where id =  ${req.params.id}`;
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log("ERROR fetching from the database:", err);
+      return;
+    }
+    console.log("Request succeeded, new data fetched", rows);
+    res.status(200).json({
+      customers: rows
+    });
+  });
 });
 
-
-router.get('/customers/:surname', function(req, res) {
+router.get("/customers/search/:surname", function(req, res) {
   // TODO: add code here
+  var sql = `select * from customers where surname like '%${
+    req.params.surname
+  }%'`;
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log("ERROR fetching from the database:", err);
+      return;
+    }
+    console.log("Request succeeded, new data fetched", rows);
+    res.status(200).json({
+      customers: rows
+    });
+  });
 });
 
-
-router.post('/customers/', function(req, res) {
-  // EXPECTED JSON Object:
-  // {
-  //   title: 'Mr',
-  //   firstname: 'Laurie',
-  //   surname: 'Ainley',
-  //   email: 'laurie@ainley.com'
-  // }
-
-  // TODO: add code here
+router.post("/customers/", function(req, res) {
+  let newData = req.body;
+  var sql = `INSERT INTO customers (title, first_name, surname, email) VALUES (
+   '${newData.title}',
+  '${newData.first_name}',
+  '${newData.surname}',
+  '${newData.email}')`;
+
+  console.log(sql);
+  db.run(sql, [], (err, rows) => {
+    if (err) {
+      console.log("ERROR fetching from the database:", err);
+      return;
+    }
+    console.log("Request succeeded, new data fetched", rows);
+    res.status(200).json({
+      result_message: "Successfuly inserted new customer."
+    });
+  });
 });
 
-
-router.put('/customers/:id', function(req, res) {
-  // EXPECTED JSON Object:
-  // {
-  //   title: 'Mr',
-  //   firstname: 'Laurie',
-  //   surname: 'Ainley',
-  //   email: 'laurie@ainley.com'
-  // }
-
-  // TODO: add code here
+router.put("/customers/:id/surname/:surname/firstname/:first_name", function(
```

Comment:
You should only be putting id in the request url. surname & firstname should be sent in the request body as JSON.